src/averageAges.js

```diff
@@ -37,7 +47,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const namesMentionedAsMother = people.map(person => person.mother);
+  const filteredWomen = people
+    .filter(person => person.sex === 'f'
+      && (!withChildren || namesMentionedAsMother.includes(person.name)));
+
+  const sumOfAge = filteredWomen
+    .map(woman => woman.died - woman.born)
+    .reduce((a, b) => a + b);
+
+  const averageAge = sumOfAge / filteredWomen.length;
```

Comment:
Here you repeat the same logic as for men. It's better to create a separate function to calculate the average age